### ISSUE TITLE
Goal doesn't update in a dashboard if the title renamed in the dashboard

### ISSUE THREAD

**Issue by cyliu0** [metabase/metabase#29633]
**Read this before creating a bug report**
- if any of the sections are incomplete, we might close the bug report since we won't be able to reproduce it
- have you checked if the bug has already been reported? As we close duplicate reports, please search on the issue tracker if the problem has been reported before
- use as many screenshots or videos as possible to describe what you're seeing, all visual representations speak for themselves

**Describe the bug**
The two cards below are the same card. The left one has been renamed when added the card to this dashboard. The right one has not been renamed. When I change the goal in the card page, the left one's goal will not be updated.

**Logs**
Please include JavaScript console logs (open the browser dev tools and check the "console" tab) as well as the server logs (Metabase logs that you can get from Settings -> Admin -> Troubleshooting -> Logs) around the time this bug occurred. For information about how to get these, consult our [bug troubleshooting guide](https://metabase.com/docs/latest/troubleshooting-guide/bugs.html)

**To Reproduce**
Refer to the bug description

**Expected behavior**
The goal should be updated even if the name of the card has been renamed in a dashboard

**Screenshots/videos**
<img width="1272" alt="image" src="https://user-images.githubusercontent.com/35356271/228411084-64fae34d-674a-469c-8f04-416d38070622.png">

**Information about your Metabase Installation:**

```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.18+10",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.18",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.18+10",
    "os.name": "Linux",
    "os.version": "4.14.294-220.533.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-02-19",
      "tag": "v0.45.3",
      "branch": "release-x.45.x",
      "hash": "070f57b"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
Basically, we rename most of the titles in many aggregation dashboards to deal with our business data in different aspects. This is a quite urgent bug for us.

**Additional context**

**Comment by paoliniluis** [comment]
If you change things in the dashboard, that won't be persisted back to the card. Card viz-settings and dashboard viz-setting are different

**Comment by cyliu0** [comment]
@paoliniluis Of course, I understand this design. But the implicit design is still a little bit weird. The dashboard goal doesn't change when the goal in the card viz-settings changed only if the dashboard card viz-settings has been modified. Can we sync parts of the card viz-settings to the dashboard card viz-settings? It will be more reasonable. Especially for users using the same cards in different dashboards with different dashboard card names.

**Comment by paoliniluis** [comment]
I don’t think that’s how we will proceed in the product unfortunately, at least in the short term
